fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/shears/DispenserBlockMixin.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.item.shears;
+
+import java.util.Map;
+
+import com.llamalad7.mixinextras.injector.ModifyReturnValue;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+
+import net.minecraft.block.DispenserBlock;
+import net.minecraft.block.dispenser.DispenserBehavior;
+import net.minecraft.block.dispenser.ShearsDispenserBehavior;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.impl.item.ShearsHelper;
+
+@Mixin(DispenserBlock.class)
+public abstract class DispenserBlockMixin {
+	@Shadow
+	@Final
+	private static Map<Item, DispenserBehavior> BEHAVIORS;
+
+	@ModifyReturnValue(at = @At("TAIL"), method = "getBehaviorForItem")
+	private DispenserBehavior registerShearsBehavior(DispenserBehavior original, ItemStack stack) {
+		// allows anything in fabric:shears to have the dispenser behavior of shears,
+		// but only if there isn't a dispenser behavior already registered
+		Item item = stack.getItem();
+
+		if (ShearsHelper.isShears(stack) && !BEHAVIORS.containsKey(item)) {
+			DispenserBehavior behavior = new ShearsDispenserBehavior();
+			BEHAVIORS.put(item, behavior);
```

Comment:
Wont this still be regsistered if the tags are reloaded to no longer contain this item as shears?